pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -169,6 +169,32 @@ func Pod(source *v1alpha1.CatalogSource, name string, image string, saName strin
 		},
 	}
 
+	// Override scheduling options if specified
+	if source.Spec.GrpcPodConfig != nil {
+		grpcPodConfig := source.Spec.GrpcPodConfig
+
+		// Override node selector
+		if grpcPodConfig.NodeSelector != nil {
+			pod.Spec.NodeSelector = make(map[string]string, len(grpcPodConfig.NodeSelector))
+			for key, value := range grpcPodConfig.NodeSelector {
+				pod.Spec.NodeSelector[key] = value
+			}
```

Comment:
non blocking nit: why is copying the map necessary here? Why not make `grpcPodConfig.NodeSelector` and `pod.Spec.NodeSelector` point to the same thing?

Reply:
I thought I'd be defensive here because the nodeselector map is passed by reference. If possible (thought probably unlikely) that you could have out of band changes make it into the pod spec (e.g. if something else changes the map). So, I thought it safer to copy it over and avoid possible mysterious bugs.

Reply:
okay that's fair